storage/spanstore/memory/memory.go

```diff
@@ -46,13 +48,60 @@ func NewStore() *Store {
 		traces:     map[model.TraceID]*model.Trace{},
 		services:   map[string]struct{}{},
 		operations: map[string]map[string]struct{}{},
+		deduper:    adjuster.SpanIDDeduper(),
 	}
 }
 
 // GetDependencies returns dependencies between services
 func (m *Store) GetDependencies(endTs time.Time, lookback time.Duration) ([]model.DependencyLink, error) {
-	// TODO - go through all traces and build dependencies between start and end time
-	return []model.DependencyLink{}, nil
+	deps := map[string]*model.DependencyLink{}
+	m.Lock()
+	defer m.Unlock()
+	startTs := endTs.Add(-1 * lookback)
+	for _, orig := range m.traces {
+		// SpanIDDeduper never returns an err
+		trace, _ := m.deduper.Adjust(orig)
+		if m.traceIsBetweenStartAndEnd(startTs, endTs, trace) {
+			for _, s := range trace.Spans {
+				parentSpan := m.findSpan(trace, s.ParentSpanID)
+				if parentSpan != nil {
+					depKey := parentSpan.Process.ServiceName + "&&&" + s.Process.ServiceName
```

Comment:
I would skip the count if `parentSpan.Process.ServiceName == s.Process.ServiceName`?